File: BackEnd/model/Papermodel.js
const mongoose = require("mongoose");

const paperSchema = new mongoose.Schema({
  title: {
    type: String,
    required: true,
  },
  authors: {
    type: [String],
    required: true,
  },
  year: {
    type: Number,
    required: true,
    default: 0,
  },
  pdfUrl: {
    type: String,
    required: true,
  },
  // Store uploader name or identifier as plain string (optional)
  uploadedBy: {
    type: String,
    default: "",
  },
  category: {
    type: String,
    default: "",
  },
  averageRating: {
    type: Number,
    default: 0,
    required: true,
  },
  reviewCount: {
    type: Number,
    default: 0,
    required: true,
  },
  createdAt: {
    type: Date,
    default: Date.now,
    required: true,
  },
  updatedAt: {
    type: Date,
    default: Date.now,
    required: true,
  },
});

module.exports = mongoose.model("Paper", paperSchema);

File: app/src/main/java/com/example/intellireview_research_paper/model/paperModel.kt
package com.example.intellireview_research_paper.model

import com.google.gson.annotations.SerializedName

data class paperModel(
    @SerializedName("_id")
    val paperId: String? = null,

    val title: String? = null,
    val authors: List<String> = emptyList(),
    val year: Int = 0,
    val pdfUrl: String? = null,
    val uploadedBy: String? = null,
    val category: String? = null,
    val averageRating: Double = 0.0,
    val reviewCount: Int = 0,
    val createdAt: String? = null,
    val updatedAt: String? = null
)
